static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -0,0 +1,250 @@
+  addIfCondition: (groupIndex: number, type: string) => void;
+  id: number;
+  onDelete: () => void;
+  removeIfCondition: (groupIndex: number, conditionIndex: number) => void;
+  updateIfCondition: (
+    groupIndex: number,
+    conditionIndex: number,
+    comparison: any
+  ) => void;
+  updateIfLogicType: (groupIndex: number, logicType: DataConditionGroupLogicType) => void;
+}
+
+function ActionFilterBlock({
+  id,
+  onDelete,
+  addIfCondition,
+  removeIfCondition,
+  updateIfCondition,
+  updateIfLogicType,
+}: ActionFilterBlockProps) {
+  const {state} = useAutomationBuilderContext();
+  const actionFilterBlock = state.actionFilters[id];
+
+  return (
+    <IfThenWrapper key={id}>
+      <Step>
+        <Flex column gap={space(0.75)}>
+          <Flex justify="space-between">
+            <StepLead>
+              {tct('[if: If] [selector] of these filters match', {
+                if: <Badge />,
+                selector: (
+                  <EmbeddedWrapper>
+                    <EmbeddedSelectField
+                      styles={{
+                        control: (provided: any) => ({
+                          ...provided,
+                          minHeight: '21px',
+                          height: '21px',
+                        }),
+                      }}
+                      inline={false}
+                      isSearchable={false}
+                      isClearable={false}
+                      name={`actionFilters.${id}.logicType`}
+                      required
+                      flexibleControlStateSize
+                      options={FILTER_MATCH_OPTIONS}
+                      size="xs"
+                      value={actionFilterBlock?.logicType}
+                      onChange={value => updateIfLogicType(id, value)}
+                    />
+                  </EmbeddedWrapper>
+                ),
+              })}
+            </StepLead>
+            <DeleteButton
+              aria-label={t('Delete If/Then Block')}
+              size="sm"
+              icon={<IconDelete />}
+              borderless
+              onClick={onDelete}
+            />
+          </Flex>
+          <RuleNodeList
+            placeholder={t('Filter by...')}
+            group={`actionFilters.${id}`}
+            conditions={actionFilterBlock?.conditions || []}
+            onAddRow={type => addIfCondition(id, type)}
+            onDeleteRow={index => removeIfCondition(id, index)}
+            updateCondition={(index, comparison) =>
+              updateIfCondition(id, index, comparison)
+            }
+          />
+        </Flex>
+      </Step>
+      <Step>
+        <StepLead>
+          {tct('[then:Then] perform these actions', {
+            then: <Badge />,
+          })}
+        </StepLead>
+        {/* TODO: add actions dropdown here */}
+      </Step>
+    </IfThenWrapper>
+  );
+}
+
+const PurpleTextButton = styled(Button)`
+  color: ${p => p.theme.purple300};
+  font-weight: normal;
+  padding: 0;
+`;
```

Comment:
Hmm... we don't have a variant like this on the `Button` component?

Reply:
I don't think so 😔  the closest thing I see in the stories is the `link` variant but that's not right for this scenario